### PR TITLE
[Dashboard] Add cluster utilization graph

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

Screen Shot 2023-01-24 at 1 01 25 AM

This PR adds the cluster utilization page in the landing view

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @scottsun94 Btw, can we make the card size a bit bigger at the landing page? I think it'd be great if we can show all legends from without scrolling down (rn, only 2 legends are visible for the cluster utilization view)

**Comment by scottsun94** [comment]
> cc @scottsun94 Btw, can we make the card size a bit bigger at the landing page? I think it'd be great if we can show all legends from without scrolling down (rn, only 2 legends are visible for the cluster utilization view)

Good point. Actually, @alanwguo and I discussed it too and we agreed that we should make these summary cards taller since metrics will stay in their own page. 

@alanwguo Will you make the change in another PR or @rkooo567 should just change it here?
It's hard for me to define a height now. I'll give feedback later if needed.

**Comment by rkooo567** [comment]
@alanwguo why don't you just push changes directly to this PR?

**Comment by alanwguo** [comment]
> @alanwguo why don't you just push changes directly to this PR?

okay. I will create the new metrics page first and then we can merge with this PR and adjust the height.

File: dashboard/modules/metrics/grafana_dashboard_factory.py [hunk 1]
```diff
@@ -499,17 +548,26 @@ def _generate_grafana_panels() -> List[dict]:
...
+        template["stack"] = panel.stack
         panels.append(template)
     return panels
 
 
 GLOBAL_FILTERS = ['SessionName="$SessionName"']
 
 
+def gen_incrementing_alphabets(length):
+    # 65: ascii code of 'A'.
```
Comment (alanwguo) [inline dashboard/modules/metrics/grafana_dashboard_factory.py:561]: should we assert if we go over 26?

File: dashboard/modules/metrics/grafana_dashboard_factory.py [hunk 2]
```diff
@@ -361,8 +365,53 @@ def max_plus_pending(max_resource, pending_resource):
...
+        ],
+        fill=0,
+        stack=False,
+    ),
 ]
 
+ids = []
+for panel in GRAFANA_PANELS:
+    ids.append(panel.id)
+assert len(ids) == len(set(ids)), ids
```
Comment (alanwguo) [inline dashboard/modules/metrics/grafana_dashboard_factory.py:413]: Can we add a message "Duplicate id found"?

**Comment by ericl** [comment]
A couple request:
- Could we add a "View all metrics" link similar to the "View all nodes" one?
- Could we show the logical resource utilization instead of the physical one? The latter may confuse new users, if they launch a bunch of tasks/actors and see no physical usage.

**Comment by scottsun94** [comment]
> A couple requests:
> 
> * Could we add a "View all metrics" link similar to the "View all nodes" one?
> * Could we show the logical resource utilization instead of the physical one? The latter may confuse new users, if they launch a bunch of tasks/actors and see no physical usage.

1. Yes to the first one. @alanwguo and discussed that before.
2. I want to push back on this one. I don't think that the logical resource concept itself (and the distinction between it and physical resources) is well understood by users, especially new users. During the previous research, whenever we show users "logical resource" charts, they cannot really tell what they mean and often confuse them with physical resources.

**Comment by ericl** [comment]
Ok--- after thinking about it more I'm ok with showing physical utilization. The main reason is that it's hard to show logical utilization of memory and disk, and it would require more UI changes to show those in a reasonable way.

Let's make sure to make the link out to more metrics prominent...

**Comment by scottsun94** [comment]
SG. @alanwguo already added the "View all metrics" button in another PR.
Screen Shot 2023-01-26 at 12 37 29 PM

To make the button more prominent, could we add more spacing between grafana chart and the button so that it stands out more?
Screen Shot 2023-01-26 at 12 39 33 PM

**Comment by rkooo567** [comment]
@alanwguo addressed.

For other parts (go metric tab + bigger card) will be in other PR, so I won't do anything

**Comment by alanwguo** [comment]
I'll increase the size of the metrics cards + do the follow-ups of #31946 in this PR

**Comment by alanwguo** [comment]
Screenshot 2023-01-27 at 12 06 13 PM
Screenshot 2023-01-27 at 12 06 40 PM
Screenshot 2023-01-27 at 12 13 09 PM

**Review by alanwguo — APPROVED** [review]

**Comment by scottsun94** [comment]
Thanks! LGTM.

**Pull request merged**